From ccc73c9fc81b79c24603923efcb00aab2a1d4353 Mon Sep 17 00:00:00 2001
From: Philip, Avinash <avinashphilip@ti.com>
Date: Wed, 8 Aug 2012 18:17:11 +0530
Subject: [PATCH 2/2] PWM: ecap: Resets the PWM output to low on stop

When PWM is stopped, the output on the PWM pin is not reset to low. This
can result in un-predictable output (example pwm brightness goes to 100%
if the line is high upon stop).
This patch fixes by disabling APWM mode on stop & enable it on start.

Signed-off-by: Philip, Avinash <avinashphilip@ti.com>
---
 drivers/pwm/ecap.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/pwm/ecap.c b/drivers/pwm/ecap.c
index a75ef36..56cba1a 100644
--- a/drivers/pwm/ecap.c
+++ b/drivers/pwm/ecap.c
@@ -77,7 +77,7 @@ static int ecap_pwm_stop(struct pwm_device *p)
 
 	spin_lock_irqsave(&ep->lock, flags);
 	v = readw(ep->mmio_base + CAPTURE_CTRL2_REG);
-	v &= ~ECTRL2_CTRSTP_FREERUN;
+	v &= ~(ECTRL2_CTRSTP_FREERUN | ECTRL2_MDSL_ECAP);
 	writew(v, ep->mmio_base + CAPTURE_CTRL2_REG);
 	spin_unlock_irqrestore(&ep->lock, flags);
 
@@ -103,7 +103,7 @@ static int ecap_pwm_start(struct pwm_device *p)
 
 	spin_lock_irqsave(&ep->lock, flags);
 	v = readw(ep->mmio_base + CAPTURE_CTRL2_REG);
-	v |= ECTRL2_CTRSTP_FREERUN;
+	v |= (ECTRL2_CTRSTP_FREERUN | ECTRL2_MDSL_ECAP);
 	writew(v, ep->mmio_base + CAPTURE_CTRL2_REG);
 	spin_unlock_irqrestore(&ep->lock, flags);
 	set_bit(FLAG_RUNNING, &p->flags);
-- 
1.7.0.4

